_posts/2021-12-15-dd_00004_logic_gates_2.md

```diff
@@ -0,0 +1,369 @@
+
+And that's why an AND gate is constructed using a NAND + Inverter. An AND gate
+therefore consists of six transistors. If you still don't believe us: try
+designing a CMOS AND gate with less than six transistors, and where all PMOS
+transistors are in the pull-up network, and all NMOS transistors are in the
+pull-down network. (hint: you will successfully fail).
+
+What about the **OR** gate? We leave that task to the reader, it's not that
+difficult if you understood how the AND gate is made! (you can also find the
+solution in the next section ;)
+
+## Complex circuit
+
+Let's put it all together - quite literally. We will show you how the transistor
+schematic of the circuit from last post's challenge looks like. Here is the
+circuit again:
+
+<figure class="img-center" style="width:90%">
+  <img src="/assets/images/dd-00003-excercise.jpg" alt="">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
+</figure>
+
+A little warning: don't be scared, this is the transistor implementation of the
+same circuit:
+
+<figure class="img-center" style="width:98%">
+  <img src="/assets/images/dd-00004-complex.jpg" alt="complex CMOS circuit">
+  <!-- <figcaption class="figure-caption text-center">Microchips</figcaption> -->
+</figure>
+
+To make it easier for you, we have circled the logic gates in different colors.
+We can recognize the CMOS transistors of the AND, OR and NAND gates. The CMOS
+XOR we didn't show you earlier, but from the circuit you can see that it looks
+much more complex than the other gates. Overall, this circuit made up of just
+four logic gates already consumes 28 individual transistors! Drawing this by
+hand is still somewhat manageble. But think of a big processor: those beasts
+consist of _millions_ of logic gates (_not_ transistors). And since each gate is
+made up of multiple transistors itself, you can imagine, that designing an
+entire CPU by hand is straight up infeasible nowadays! That's why we have to use
+the help of **Electronic Design Automation** (EDA), i.e. software to automate
+the design of digital circuits, when we build such complex chips. These tools
+will do the mapping from logic gates to transistors fully automatic! If you are
+interested, we can do a post about EDA, what tools there are and how they work;
+let us know in the comments (we might do it anyways ;)
+
+## Physical Layout
+
+One final note. Remember when we said in our last post, that the symbols we use
+for drawing logic gates were not how the actual gate would look in hardware?
+Well, even the transistor diagrams we showed you in this post don't actually
+depict how the real hardware looks, but they are already pretty close! In case
+you are wondering, here is how a NAND _actually_ looks on the final chip:
+
+<figure class="img-center" style="width:30%">
+  <img src="/assets/images/dd-00004-cmos-nand-layout.svg" alt="CMOS NAND Layout">
+  <figcaption class="text-center">
+   (source: Wikipedia)
```

Comment:
I always forget ... Is it sufficient to say it is from Wikipedia or is the full path necessary? We used some generator for that last time

Reply:
Actually, this image is free of any license it seems ("public domain"): https://commons.wikimedia.org/wiki/File:CMOS_NAND_Layout.svg

But yes, when there is a license, we can use one of these generators.